### PR TITLE
Added the Scipy test suite

### LINKED ISSUE #145: Add the scipy suite of tests
The scipy modules are untested as of now. Create a suite of unit tests for these.

### PR DESCRIPTION
**Pull request by bb511**
#### Description of Work

Fixes #145 

Added the scipy suite of tests and fixed minor mistakes in the code.

#### Testing Instructions

1. Check if the tests pass on travis.
2. Check if the code is ok.

### CODE REVIEW

**Comment by Anders-Markvardsen** [comment]
Observe that on Windows when I run: scipy/tests/test_nist_def.py, scipy/tests/test_neutron_def.py and scipy/tests/test_prepare_data.py they all 3 passes. 

But when I run fitbenchmarking/fitting/scipy/tests/test_main.py I get

```
======================================================================
ERROR: test_execute_fit_cost_function_least_squares (__main__.ScipyTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Midlertiddig\fitbenchmarking\fitbenchmarking\fitting\scipy\tests\test_main.py", line 136, in test_execute_fit_cost_function_least_squares
    minimizer, cost_function)
  File "C:\Midlertiddig\fitbenchmarking\fitbenchmarking\fitting\scipy\main.py", line 148, in execute_fit
    p0=initial_params, method=minimizer, maxfev=500)
  File "C:\MantidInstall\bin\lib\site-packages\scipy\optimize\minpack.py", line 581, in curve_fit
    res = leastsq(func, p0, args=args, full_output=1, **kw)
TypeError: leastsq() got an unexpected keyword argument 'method'
```

In this testing terminal have scipy version 0.16.0 installed

File: fitbenchmarking/fitting/scipy/prepare_data.py [hunk 1]
```diff
@@ -78,5 +78,5 @@ def apply_constraints(start_x, end_x, data_x, data_y, data_e):
     data_x = np.copy(data_x[start_idx:end_idx])
     data_y = np.copy(data_y[start_idx:end_idx])
     data_e = np.copy(data_e[start_idx:end_idx])
-    data_e[data_e == 0] = 0.000001
+    data_e[data_e == 0] = 0.00000001
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/scipy/prepare_data.py]: Optionally consider explain why you reduced this value

**Comment by bb511** [comment]
This issue is due to mantidpython having an old version of scipy. For future users, the problem and a fix to it is described here https://github.com/mantidproject/fitbenchmarking#fitbenchmarking-scipy . Unfortunately, this cannot be fixed from within the code.

**Comment by Anders-Markvardsen** [comment]
@bb511 updated readme accordingly, corrected a minor typo scipy upgrade instructions and with that all works. Don't hesitate to comment on modifications to readme otherwise looks good

**Comment by bb511** [comment]
@Anders-Markvardsen Looks good to me!

**Comment by bb511** [comment]
@Anders-Markvardsen with regards to travis failing just now, that is because it could not download mantid from the official servers. This sometimes happen rather infrequently in the past, but has increased in frequency recently.

**Comment by Anders-Markvardsen** [comment]
and thanks for the comment about travis

**Pull request merged**